engine/src/org/pentaho/di/trans/steps/mapping/MappingMeta.java

```diff
@@ -712,75 +707,139 @@ public static final synchronized TransMeta loadMappingMeta( MappingMeta mappingM
     return mappingTransMeta;
   }
 
-  public void check( List<CheckResultInterface> remarks, TransMeta transMeta, StepMeta stepMeta,
-    RowMetaInterface prev, String[] input, String[] output, RowMetaInterface info, VariableSpace space,
-    Repository repository, IMetaStore metaStore ) {
+  /*
+   * loads transformation from local file system or from repository if failed and connected to repository
+   */
+  public static TransMeta loadTransFromAnyPath( String realFilename, Repository repo, IMetaStore metaStore )
```

Comment:
I'm not sure I agree with this at all. This seems to me to be flawed reasoning and potentially a security vulnerability. It should not grab something off the file system if you attempt to load it from the repository and it fails. That's just bad. The issue to resolve is that ${Internal.Entry.Current.Directory} isn't being resolved when connected to a repository and the transformation/job are stored in the repository. Please fix that.